File: app/src/main/java/org/schabi/newpipe/util/text/HashtagLongPressClickableSpan.java
package org.schabi.newpipe.util.text;

import android.content.Context;
import android.view.View;

import androidx.annotation.NonNull;

import org.schabi.newpipe.util.NavigationHelper;
import org.schabi.newpipe.util.external_communication.ShareUtils;

final class HashtagLongPressClickableSpan extends LongPressClickableSpan {

    @NonNull
    private final Context context;
    @NonNull
    private final String parsedHashtag;
    private final int relatedInfoServiceId;

    HashtagLongPressClickableSpan(@NonNull final Context context,
                                  @NonNull final String parsedHashtag,
                                  final int relatedInfoServiceId) {
        this.context = context;
        this.parsedHashtag = parsedHashtag;
        this.relatedInfoServiceId = relatedInfoServiceId;
    }

    @Override
    public void onClick(@NonNull final View view) {
        NavigationHelper.openSearch(context, relatedInfoServiceId, parsedHashtag);
    }

    @Override
    public void onLongClick(@NonNull final View view) {
        ShareUtils.copyToClipboard(context, parsedHashtag);
    }
}

File: app/src/main/java/org/schabi/newpipe/util/text/HashtagLongPressClickableSpan.kt
package org.schabi.newpipe.util.text

import android.content.Context
import android.view.View
import org.schabi.newpipe.util.NavigationHelper
import org.schabi.newpipe.util.external_communication.ShareUtils
import org.schabi.newpipe.views.LongPressClickableSpan

internal class HashtagLongPressClickableSpan(
    private val context: Context,
    private val parsedHashtag: String,
    private val serviceId: Int
) : LongPressClickableSpan() {

    override fun onClick(view: View) {
        NavigationHelper.openSearch(context, serviceId, parsedHashtag)
    }

    override fun onLongClick(view: View) {
        ShareUtils.copyToClipboard(context, parsedHashtag)
    }
}
